pkg/generate/app/cmd/newapp.go

```diff
@@ -325,10 +337,17 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 				if err != nil {
 					return nil, fmt.Errorf("can't build %q: %v", ref.Input(), err)
 				}
-				strategy, source, err := app.StrategyAndSourceForRepository(ref.Input().Uses)
+				strategy, source, err := app.StrategyAndSourceForRepository(ref.Input().Uses, app.PerfectMatchWeightedResolver{
```

Comment:
Cesar why did we change the weight?  Or was it just copied *before* new-app existed.